include/nbl/ui/IWindowXCB.h

```diff
@@ -0,0 +1,88 @@
+
+#include <xcb/xproto.h>
+
+namespace nbl::ui
+{
+
+class NBL_API2 IWindowXCB : public IWindow
+{
+    protected:
+        virtual ~IWindowXCB() = default;
+        inline IWindowXCB(SCreationParams&& params) : IWindow(std::move(params)) {}
+
+    public:
+        using IWindow::IWindow;
+
+        const void* getNativeHandle() const { return nullptr; }
+        virtual xcb_window_t getXcbWindow() const = 0;
+        virtual xcb_connection_t* getXcbConnection() const = 0;
+
+        virtual bool setWindowSize_impl(uint32_t width, uint32_t height) = 0;
+        virtual bool setWindowPosition_impl(int32_t x, int32_t y) = 0;
+        virtual bool setWindowRotation_impl(bool landscape) = 0;
+        virtual bool setWindowVisible_impl(bool visible) = 0;
+        virtual bool setWindowMaximized_impl(bool maximized) = 0;
+
+        static XCBConnection::MotifWmHints fetchMotifMWHints(IWindow::E_CREATE_FLAGS flags) {
+            core::bitflag<XCBConnection::MotifFlags> motifFlags(XCBConnection::MWM_HINTS_NONE);
+            core::bitflag<XCBConnection::MotifFunctions> motifFunctions(XCBConnection::MWM_FUNC_NONE);
+            core::bitflag<XCBConnection::MotifDecorations> motifDecorations(XCBConnection::MWM_DECOR_NONE);
+            motifFlags |= XCBConnection::MWM_HINTS_DECORATIONS;
+
+            if (flags & IWindow::ECF_BORDERLESS) {
+                motifDecorations |= XCBConnection::MWM_DECOR_ALL;
+            } else {
+                motifDecorations |= XCBConnection::MWM_DECOR_BORDER;
+                motifDecorations |= XCBConnection::MWM_DECOR_RESIZEH;
+                motifDecorations |= XCBConnection::MWM_DECOR_TITLE;
+
+                // minimize button
+                if(flags & IWindow::ECF_MINIMIZED) {
+                    motifDecorations |= XCBConnection::MWM_DECOR_MINIMIZE;
+                    motifFunctions |= XCBConnection::MWM_FUNC_MINIMIZE;
+                }
+                
+                // maximize button
+                if(flags & IWindow::ECF_MAXIMIZED) {
+                    motifDecorations |= XCBConnection::MWM_DECOR_MAXIMIZE;
+                    motifFunctions |= XCBConnection::MWM_FUNC_MAXIMIZE;
+                }
+
+                // close button
+                motifFunctions |= XCBConnection::MWM_FUNC_CLOSE;
+            }
+
+            if(motifFunctions.value != XCBConnection::MWM_FUNC_NONE) {
+                motifFlags |= XCBConnection::MWM_HINTS_FUNCTIONS;
+                motifFunctions |= XCBConnection::MWM_FUNC_RESIZE;
+                motifFunctions |= XCBConnection::MWM_FUNC_MOVE;
+            } else {
+                motifFunctions = XCBConnection::MWM_FUNC_ALL;
+            }
+
+            XCBConnection::MotifWmHints hints;
+            hints.flags = motifFlags.value;
+            hints.functions = motifFunctions.value;
+            hints.decorations = motifDecorations.value;
+            hints.input_mode = 0;
+            hints.status = 0;
+            return hints;
+
+        }
```

Comment:
all this stuff you want to push into `CWindowXCB`, so that you may remove any XCB and X11 header includes from headers in `include/nbl/ui`